addons/spreadsheet/static/src/pivot/plugins/pivot_ui_plugin.js

```diff
@@ -14,26 +14,33 @@ const { DateTime } = luxon;
  * Convert pivot period to the related filter value
  *
  * @param {import("@spreadsheet/global_filters/plugins/global_filters_core_plugin").RangeType} timeRange
- * @param {string} value
+ * @param {string|number} value
  * @returns {object}
  */
 function pivotPeriodToFilterValue(timeRange, value) {
     // reuse the same logic as in `parseAccountingDate`?
-    const yearOffset = (value.split("/").pop() | 0) - DateTime.now().year;
+    if (typeof value === "number") {
+        value = `${value}`;
```

Comment:
this seems slightly better than using Number.toString() on chrome but on firefox "il y a pas match"
![image](https://github.com/odoo/odoo/assets/10333790/9b29a077-b59b-4710-a8ba-81d3a01bd6c7)

Reply:
That being said, because I talk then I think, this piece of code is not stress-called so it might not matter that much.